zip-liberated-payments.rst

```diff
@@ -0,0 +1,410 @@
+
+Bob receives a message, which looks as follows:
+
+    This message contains a Zcash payment.
+    Click the following link to view it and receive the funds using your Zcash wallet app: 
+    
+https://pay.withzcash.com:65536/v1#amount=1.23&desc=Payment+for+foo&key=...
+    If you do not yet have a Zcash wallet app, see: https://z.cash/wallets
+
+Bob clicks the link. His Zcash mobile wallet app (which is already installed and has configured itself as a handler for URLs of that form) shows up, and tells Bob that a payment of 1.23 ZEC awaits him. The wallet app confirms the existence of the pertinent transactions on the blockchain, and then offers to finalize the payment. Bob clicks the “Finalize” button, and his wallet app generates a transaction moving the money to his own address (using the extra 0.0001 ZEC he has received to pay the transaction fee). When this transaction is confirmed on-chain, Alice’s and Bob’s wallets both indicate that the payment is finalized, and thus Bob can send the funds to other parties.
+
+
+Motivation
+==========
+
+This proposal enables sending of funds without exposing users to the notion of payment addresses and their secure distribution. Instead, funds can be sent using any pre-existing communication channel, by a single message sent unidirectionally from the sender to the recipient. This message conveys all the information needed to obtain control of the funds, compactly expressed as a textual URI.
+
+Consequently, all functionality related to contact discovery and secure-channel establishment can be delegated to the message app(s) with which the user is already familiar, and in which the user has already established communication channels to many of their contacts.
+
+Moreover, funds can be sent to users who have not yet installed wallet software and who do not yet have a payment address. The recipient can collect the funds at any later time by installing suitable software.
+
+The proposal is complementary to ZIP 321 [#zip321]_, which will standardize *Payment Request URIs* using which the payment *recipient* can convey their persistent payment address to the *sender*, for subsequent fund transfers (to be done using the normal on-chain mechanism rather than the encapsulated payments described in the current proposal).
+
+
+Requirements
+============
+
+This proposal’s specification of Payment-Encapsulating URIs, and the intended protocols for using them, is meant to fulfill the following requirements:
+
+* The protocol must not require the sender of a payment to stay online until the recipient receives the URI (let alone finalizes the payment).
+
+* The URIs should be short, for convenient rendering and to maximize compatibility with length-limited messaging platforms.
```

Comment:
```suggestion
* The URIs should be short for convenient rendering and to maximize compatibility with length-limited messaging platforms.
```

Reply:
To me, a comma maybe doesn't go here because the sentence doesn't make sense when the clause is removed:

> The URIs should be short and to maximize compatibility with length-limited messaging platforms.

This suggests that the "convenient rendering" goes along with the other thing.

Perhaps this statement could be reworded to something like:

For convenient rendering, the URIs should be short. This also maximizes compatibility with length-limited messaging platforms.